packages/documentation/src/stories/patterns/forms/forms.docs.mdx

```diff
@@ -0,0 +1,97 @@
+
+## Vertical spacing
+To add space between input, you can use [margin utilities](/?path=/docs/utilities-spacing--docs).
+
+<Canvas of={FormPatternsStories.VerticalSpacing} sourceState="shown" />
+
+## Rows'n'cols
+
+If you want a form field to span the whole available width, you don't need a row nor a col. You
+only need those, if you want to place two things next to each other.
+
+Simply use `.row` and `.col` to have the width distributed evenly between all inputs.
+To make it responsive, you can add `.row-cols-*` classes to the row (see [Grid page](/?path=/docs/foundations-layout-grid--docs#row-columns)).
+
+<Canvas of={FormPatternsStories.RowSimple} sourceState="shown" />
+
+For finer adjustment, you can also use the `.col-*` classes (see [Columns page](/?path=/docs/foundations-layout-columns--docs)) on your columns. For example for a 2/1 ratio, given that a row has twelve columns in total,
+just add `.col-8` and `.col-4` to you two columns respectively.
+
+<Canvas of={FormPatternsStories.ColSimple} sourceState="shown" />
+
+## Fields with custom width
+
+Some fields may not fit into the twelve row schema, they need a custom width. The PLZ field may
+be one of those cases. Two `.col` next to each other will have equal width. If one of the cols has the
+`.col-auto` class, this col is sized based on it's content. The other row is stretched to the container
+width. With this setup, we can define a max-width on the input field to shrink it.
+
+<Canvas of={FormPatternsStories.CustomWidth} sourceState="shown" />
+
+## Form buttons
+
+Buttons, often at the end of the form are right aligned. We can achieve this with the classes `d-flex flex-row-reverse`. This will ensure that the primary button is focused first (because of the markup order) but is
+still aligned to the right. To get some margin between the button we can use the `.gap-mini` class. Note that for the buttons, we don't need any row/col structure.
+
+<Canvas of={FormPatternsStories.Buttons} sourceState="shown" />
+
+## Floating labels
+
+For internet applications, bigger input fields with floating labels are the Post way of creating
+forms. The particularity of these fields, is that the `<label>` and `<input>`
+elements are inverted and wrapped in a `.form-floating` div.
+In addition, the input field needs a placeholder attribute, preferably equal to a single space.
+
+<Canvas of={FormPatternsStories.FloatingLabels} sourceState="shown" />
+
+## Validation messages
+
+Validation messages are placed directly after the `<input>`
+(or after the `<label>` in the case of floating label). Showing and hiding of these messages is typically controlled by
+adding and removing validation classes on the input:
+
+* The message needs the class `.invalid-feedback/.valid-feedback`
+* The form needs the class `.was-validated`. With this class you can manage when to display errors. Usually, errors should not be
+shown on untouched fields.
+ <br/>Or
```

Comment:
In the story there are both `.was-validated` and `.is-invalid/.is-valid` which can be misleading